bson/bsoncodec/bsoncodec.go

```diff
@@ -123,6 +123,16 @@ type DecodeContext struct {
 	// Ancestor is a bson.M, BSON embedded document values being decoded into an empty interface
 	// will be decoded into a bson.M.
 	Ancestor reflect.Type
+
+	// DocumentDecodeType will decode embedded documents into the defined type, rather than into primitive.D. This is a
+	// work-around for custom typing.
```

Comment:
We should make this comment more descriptive to help users understand when and how to use this configuration.

E.g.
```go
// DefaultDocumentType configures the default Go type to unmarshal top-level and nested BSON documents
// into. The DefaultDocumentType is only used if the Decoder can't determine type information based on
// the Decode output parameter (e.g. is type "interface{}" or "map[string]interface{}"). Setting
// DefaultDocumentType overrides the Ancestor field. If DefaultDocumentType is set to a type that a
// BSON document cannot be unmarshaled into (e.g. "string"), unmarshalling will result in an error.
```

Reply:
Done, with a few tweaks.  What are your thoughts?

```go
// DocumentType specifies the Go type to decode top-level and nested BSON documents into. In particular, the
// usage for this field is restricted to data typed as "interface{}" or "map[string]interface{}". If DocumentType is
// set to a type that a BSON document cannot be unmarshaled into (e.g. "string"), unmarshalling will result in an
// error. DocumentType overrides the Ancestor field.
```